src/apps/dictionaries/redux/actions.ts

```diff
@@ -389,12 +390,26 @@ export const recursivelyAddConceptsToDictionaryAction = (
     dispatch(
       startAction(indexedAction(ADD_CONCEPTS_TO_DICTIONARY, actionIndex))
     );
-
-    const referencesToAdd = await recursivelyFetchToConcepts(
-      sourceUrl,
-      concepts.map(concept => concept.id),
-      updateProgress
-    );
+    // const referencesToAdd = await recursivelyFetchToConcepts(
+     // concepts.map(concept => concept.id),
+     // updateProgress,
+    //  false,
+     // sourceUrl
+   // );
+
+   const referencesToAdd = await recursivelyFetchToConcepts(
+  '/sources/CIEL/',
+   concepts.map(concept => concept.id),
+  updateProgress
+  );
+
+    const groupedConcepts = groupBy(concepts, "source");
+    console.log(groupedConcepts)
+    // const referencesToAdd = await Object.entries(groupedConcepts).map(([source, concepts]) => recursivelyFetchToConcepts(
+    //  source,
+    //  concepts.map((concept: { id: any; }) => concept.id),
+    // updateProgress,1,
+    //  ));
```

Comment:
Alright so this is very confusing, but here's what I *think* the implementation should look like:

```ts
    const groupedConcepts = groupBy(concepts, "source_url");
    const referencesToAdd = flatten(
      await Promise.all(
        Object.entries(groupedConcepts).map(([source, concepts]) =>
          recursivelyFetchToConcepts(
            source,
            concepts.map((concept) => concept.id),
            updateProgress
          )
        )
      )
    );
```

The trick here is that by calling `map()` we're building an array of... whatever the return value is. In this case, it's the return value of the call to `recursivelyFetchToConcepts()` which actually returns a `Promise<string[]>` (a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) to return an array of strings at a later point).

In the currently working version, the `await` before the `recursivelyFetchToConcepts()` waits until the call to `recursivelyFetchToConcepts()` returns. Since we're returning an array of promises, (one for each call to `recursivelyFetchToConcepts()`, i.e., one for each source), we need to wait until all those promises are finished. `Promise.all()` allows us to do this, turning an array of Promises into a promise that resolves with an array of results. We can safely `await` that.

Finally, I used `lodash`'s `flatten` function to transform the return time from `string[][]` (an array of arrays of strings) to a single array, that should look much like the one that existed before all these changes.